### PR TITLE
feat: first person pointerlock controls

### PR DESCRIPTION
**Pull request by okorion**
This merge request introduces the implementation of first-person controls using the Pointer Lock API. The main focus of this feature is to enable seamless camera rotation and movement in 3D environments, which is essential for first-person games or applications. The following updates have been made:

- **Pointer Lock Integration**: The Pointer Lock API has been integrated to capture mouse movement for smooth yaw and pitch adjustments.

- **Keyboard Input Handling**: WASD keys are mapped for movement, and the spacebar is used for jumping. Key states are tracked to ensure precise control over character movement.

- **Physics Integration**: Movement is physics-driven, with parameters like gravity, jump force, and movement speed customizable via options.

- **Event Handling**: Event listeners for mousemove, keydown, keyup, and pointerlockchange are properly managed to ensure seamless control during gameplay.

- **Code Optimization**: Movement vectors (forward and side) are computed dynamically based on camera orientation to ensure accurate directional movement.

### CODE REVIEW

File: src/controls/FirstPersonPointerLockControls.ts [hunk 1]
```diff
@@ -0,0 +1,292 @@
...
+    this.object.translateY(this.eyeHeight);
+  }
+
+  /**
+   * Connects the pointer lock controls by adding event listeners.
+   */
+  connect() {
+    super.connect();
+
+    this.domElement?.addEventListener('click', this.requestPointerLock.bind(this));
```
Comment (Byongho96) [inline src/controls/FirstPersonPointerLockControls.ts:187]: bind 한 핸들러 함수 인스턴스에 정의해주시고,  그 함수를 `removeEventListener` 해주세요

File: src/controls/FirstPersonPointerLockControls.ts [hunk 2]
```diff
@@ -0,0 +1,292 @@
...
+  /**
+   * Disconnects the pointer lock controls by removing event listeners.
+   */
+  disconnect() {
+    super.disconnect();
+
+    this.domElement?.removeEventListener('click', this.requestPointerLock.bind(this));
+
+    document.addEventListener('keydown', this.onKeyDownHandler);
+    document.addEventListener('keyup', this.onKeyUpHandler);
```
Comment (Byongho96) [inline src/controls/FirstPersonPointerLockControls.ts:204]: `addEventListener`말고 `removeEventListener`로 바꿔주세요

File: src/controls/FirstPersonPointerLockControls.ts [hunk 3]
```diff
@@ -0,0 +1,292 @@
...
+    if (document.pointerLockElement === this.domElement) {
+      this.yaw -= event.movementX * this.rotateSpeed;
+      this.pitch -= event.movementY * this.rotateSpeed;
+    }
+  }
+
+  /**
+   * Handles pointer lock change events.
+   */
+  private onPointerLockChange() {
```
Comment (Byongho96) [inline src/controls/FirstPersonPointerLockControls.ts:232]: 이 함수는 지워주세요

File: src/controls/FirstPersonPointerLockControls.ts [hunk 4]
```diff
@@ -0,0 +1,292 @@
...
+
+    if (this.actionKeys.jump?.some(key => event.key === key)) {
+      keyStates.jump = false;
+    }
+  }
+
+  /**
+   * Requests pointer lock on the DOM element.
+   */
+  private requestPointerLock() {
```
Comment (Byongho96) [inline src/controls/FirstPersonPointerLockControls.ts:287]: onMouseDown으로 바꿔주세요

File: src/controls/FirstPersonPointerLockControls.ts [hunk 5]
```diff
@@ -0,0 +1,292 @@
...
+  }
+
+  /**
+   * Handles mouse movement events, adjusting pitch and yaw based on delta movements.
+   * @param event - The mouse movement event.
+   */
+  private onMouseMove(event: MouseEvent) {
+    if (document.pointerLockElement === this.domElement) {
+      this.yaw -= event.movementX * this.rotateSpeed;
+      this.pitch -= event.movementY * this.rotateSpeed;
```
Comment (Byongho96) [inline src/controls/FirstPersonPointerLockControls.ts:225]: `yaw`랑 `pitch` 변수 없이 바로 `object.roation`을 바꿔주세요

File: src/controls/FirstPersonPointerLockControls.ts (hunk 1 above)
Comment (okorion) [reply src/controls/FirstPersonPointerLockControls.ts]: 반영 커밋 올렸습니다!

File: src/controls/FirstPersonPointerLockControls.ts (hunk 2 above)
Comment (okorion) [reply src/controls/FirstPersonPointerLockControls.ts]: 반영 커밋 올렸습니다!

File: src/controls/FirstPersonPointerLockControls.ts (hunk 3 above)
Comment (okorion) [reply src/controls/FirstPersonPointerLockControls.ts]: 반영 커밋 올렸습니다!

File: src/controls/FirstPersonPointerLockControls.ts (hunk 5 above)
Comment (okorion) [reply src/controls/FirstPersonPointerLockControls.ts]: 반영 커밋 올렸습니다!

File: src/controls/FirstPersonPointerLockControls.ts [hunk 6]
```diff
@@ -0,0 +1,277 @@
...
+  // Physics options
+  eyeHeight: number;
+  jumpForce: number;
+  groundMoveSpeed: number;
+  floatMoveSpeed: number;
+  rotateSpeed: number;
+
+  // Mouse movement accumulators
+  private yaw: number = 0;
+  private pitch: number = 0;
```
Comment (Byongho96) [inline src/controls/FirstPersonPointerLockControls.ts:51]: 선언하신 변수도 없애주세요

**Review by Byongho96 — APPROVED** [review]

**Pull request merged**